corgi/tasks/pnc.py

```diff
@@ -30,4 +31,61 @@ def slow_fetch_pnc_sbom(purl: str, product_data, build_data, sbom_data) -> None:
     response.raise_for_status()
     logger.info(f"SBOM fetched for purl {purl}")
 
-    parse_pnc_sbom(response.json())
+    # Make sure the SBOM is valid
+    sbom = SbomerSbom(response.json())
+
+    # Create a build for the root component
+    root_build, _ = SoftwareBuild.objects.get_or_create(
+        build_id=sbom.components["root"]["meta_attr"]["pnc_build_id"],
+        build_type=SoftwareBuild.Type.PNC,
+    )
+
+    # Create components
+    components = {}
+    for bomref, component in sbom.components.items():
+        defaults = {"namespace": component["namespace"], "meta_attr": component["meta_attr"]}
+
+        if "related_url" in component:
+            defaults["related_url"] = component["related_url"]
+        if bomref == "root":
+            defaults["software_build"] = root_build
+
+        components[bomref], _ = Component.objects.update_or_create(
+            type=Component.Type.MAVEN,
+            name=component["name"],
+            version=component["version"],
+            release="",
+            arch="noarch",
+            defaults=defaults,
+        )
+
+    # Link dependencies
+    nodes = {}
+    # Create the root node from the SBOM's component
+    root_component = components.pop("root")
+    nodes["root"], _ = ComponentNode.objects.get_or_create(
+        type=ComponentNode.ComponentNodeType.SOURCE,
+        parent=None,
+        defaults={"obj": root_component},
+    )
```

Comment:
Let's also set the purl attribute of ComponentNode

```suggestion
    nodes["root"], _ = ComponentNode.objects.get_or_create(
        type=ComponentNode.ComponentNodeType.SOURCE,
        parent=None,
        purl=root_component.purl,
        defaults={"obj": root_component},
    )
```